src/raftstore/store/snap.rs

```diff
@@ -154,7 +161,14 @@ impl SnapFile {
 
     pub fn try_delete(&self) -> io::Result<()> {
         debug!("deleting {}", self.path().display());
-        fs::remove_file(self.path())
+        if !self.exists() {
+            return Ok(());
+        }
+        let size = try!(self.meta()).len();
+        try!(fs::remove_file(self.path()));
+        let mut size_track = self.size_track.wl();
+        *size_track = size_track.saturating_sub(size);
```

Comment:
If size > size_track, I guess we may meet some bugs for the size calculation before.